delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/projects/ProjectsConfig.scala

```diff
@@ -15,16 +25,106 @@ import pureconfig.generic.semiauto.deriveReader
   * @param pagination            configuration for how pagination should behave in listing operations
   * @param cacheIndexing         configuration of the cache indexing process
   * @param persistProgressConfig configuration for the persistence of progress of projections
+  * @param automaticProvisioning configuration for automatic provisioning of projects
   */
 final case class ProjectsConfig(
     aggregate: AggregateConfig,
     keyValueStore: KeyValueStoreConfig,
     pagination: PaginationConfig,
     cacheIndexing: CacheIndexingConfig,
-    persistProgressConfig: SaveProgressConfig
+    persistProgressConfig: SaveProgressConfig,
+    automaticProvisioning: AutomaticProvisioningConfig
 )
 
 object ProjectsConfig {
-  implicit final val projectConfigReader: ConfigReader[ProjectsConfig] =
+
+  /**
+    * Configuration of automatic provisioning of projects.
+    *
+    * @param enabled      flag signalling whether automatic provisioning is enabled
+    * @param permissions  the permissions applied to the newly provisioned project
+    * @param enabledReams the realms for which the provisioning is enabled(map of realm label to organization in which the projects for the realm should be created)
+    * @param description  the description of the created project
+    * @param apiMappings  the [[ApiMappings]] of the created project
+    * @param base         the base for the created project
+    * @param vocab        the vocab for the created project
+    */
+  final case class AutomaticProvisioningConfig(
+      enabled: Boolean,
+      permissions: Set[Permission],
+      enabledReams: Map[Label, Label],
+      description: String,
+      apiMappings: ApiMappings,
```

Comment:
I don't think that's of any use currently, since you just have apiMappings that will be added by default anyway. I would remove it if there is no current need for it, for simplicity

Reply:
There is the requirement for API mappings in the issue https://github.com/BlueBrain/nexus/issues/2520, please discuss there if you feel that they are not needed.